src/app/data-model-provider/Provider.h

```diff
@@ -85,6 +85,16 @@ class Provider : public ProviderMetadataTree
     ///      This includes cases where command handling and value return will be done asynchronously.
     ///    - returning a value other than Success implies an error reply (error and data are mutually exclusive)
     ///
+    /// Preconditions:
+    ///    - `request.path` MUST be valid: Invoke` is only guaranteed to function correctly for
+    ///      VALID paths (i.e. use `ProviderMetadataTree::AcceptedCommands` to check). This is
+    ///      because we assume ACL or flags (like timed invoke) have to happen before invoking
+    ///      this command.
```

Comment:
This should be more like:

```
    ///    - `request.path` MUST refer to a command that actually exists.  This is because in practice
    ///       callers must do ACL and flag checks (e.g. for timed invoke) before calling this function.
    ///
    ///       Callers that do not care about those checks should use `ProviderMetadataTree::AcceptedCommands`
    ///       to check for command existence.
```